native/packages/react-native-bpk-component-button/src/BpkButton-styles.js

```diff
@@ -168,12 +168,37 @@ const gradientColors = {
   disabled: [tokens.colorGray100, tokens.colorGray100],
 };
 
+const themeMappings = {
+  text: {
+    color: {
+      primary: 'textColor',
+      disabled: 'disabledTextColor',
+      selected: 'selectedTextColor',
+    },
+  },
+  gradient: {
+    primary: {
+      startColor: 'gradientStartColor',
+      endColor: 'gradientEndColor',
+    },
+    disabled: {
+      startColor: 'disabledBackgroundColor',
+      endColor: 'disabledBackgroundColor',
+    },
+    selected: {
+      startColor: 'selectedGradientStartColor',
+      endColor: 'selectedGradientEndColor',
+    },
+  },
+};
+
```

Comment:
I'm concerned that I've fallen into an over-engineering trap here. However, we do need a place where theme attributes are translated to the element they apply to. So on line 174, we're saying that the `color` style property for the `text` component uses the `textColor` theme attribute when the type is primary.